### ISSUE TITLE
Install Habitat Build 

### ISSUE THREAD

**Issue by pierzapin** [chef-partners/azuredevops-habitat#10]
I have habitat build and run steps in an Azure DevOps YAML Build definition running against a SelfHosted Linux Build Agent. I am encountering a number of errors as the build steps progress that appear to me to things the extension should be doing by default.

1. The Install step reports as complete (i.e. green) but the hab binary is not extracted to the expected location. This causes a "Unable to locate executable file: '/tmp/hab" error in any later habitat build step. I remedied this by manually exectracting and moving the binary from /hab-0.82.0-20190605214032-x86_64-linux/hab to /tmp/hab

2. The licence agreement is not accepted, therefore any hab build attempts will spin indefinitely. Again I manually triggered a hab build via ssh in order to accept the agreement and unblock use of the command in the build automation

3. And finally the build appears to start but now fails with: 

> mount: mounting /src on /hab/studios/src/src failed: No such file or directory
> ##[error]The process '/tmp/hab' failed with exit code 255

I'll attach the build yaml and screenshots of the errors encountered. Am I doing something wrong?

**Comment by pierzapin** [comment]
[azure-pipeline.yaml.txt](https://github.com/chef-partners/azuredevops-habitat/files/3408833/azure-pipeline.yaml.txt)
![Issue 3 Screen Shot ](https://user-images.githubusercontent.com/12144682/61499881-ed62eb80-aa1c-11e9-8073-c44983bf2cb8.png)
![Issue 2 Screen Shot](https://user-images.githubusercontent.com/12144682/61499977-3adf5880-aa1d-11e9-9675-0975d14212cb.png)
![Issue 1 Screen Shot](https://user-images.githubusercontent.com/12144682/61499979-3c108580-aa1d-11e9-94bd-4e88205e6a13.png)

**Comment by russellseymour** [comment]
Thanks for this @pierzapin 
I am about to release a new version of the Install task that handles installing Habitat on windows properly and sets the path properly. It has moved to `C:\ProgramData\habitat\hab.exe`

For accepting the license, the easiest way is to add an environment HAB_LICENSE variable to the build which is set to `accept`. I am reluctant to add it to the tasks as it would need to be selected on all the tasks (or at least the first one in the pipeline). By making it an environment variable it is a concious thing that needs to be done to accept the license. Would this work for you?

**Comment by pierzapin** [comment]
I appreciate you getting back to me on this, Russell. For clarity, I was having issues with builds on Linux  agents, although I guess you may have mentioned the Windows path to indicate that work is required here across platforms. 

Thanks also for explaining the env var option, I'll give that a whirl as well.

**Comment by russellseymour** [comment]
@pierzapin I apologise, I had clocked you were talking about Linux ... doh!!!

Can I ask if you are using the "Install Habitat" task before the "Habitat Build" task? That will ensure that Habitat is `/tmp/hab` which is where the task is looking for it.

**Comment by pierzapin** [comment]
Indeed, I do have "Habitat Install" as the first step in the pipeline. That step runs green, although as  I described in the initial report, I feel as if there is an extra step required in the install task, because a simple manual move of the downloaded asset to the expected path unblocked me. Has something changed in the way the habitat files are packaged into the tar up on api.bintray.com?

**Comment by russellseymour** [comment]
@pierzapin I am about to release the aforementioned version of the extension. Can you see if that fixes the issue you are seeing? If not then I will happily look at it again.

I do not think things have changed in the way that Hab is packaged.

Are you using version 3.x of the extension?
